### ISSUE TITLE
About the motivation analysis code?

### ISSUE THREAD

**Issue by ShunZuo-AI** [hustvl/MaskAdapter#11]
Congratulations on the acceptance of your paper by CVPR2025. Your work is very inspiring, especially in the analysis of motivation. I am very interested in your motivation analysis, but I am unable to understand some specific experimental settings. Can you provide the experimental code for the motivation analysis.
![Image](https://github.com/user-attachments/assets/f8471d88-7856-4009-b16e-10d9c41fba41)

**Comment by owl-10** [comment]
The current mask-based method for open-vocabulary segmentation is mainly divided into two steps: extracting class-agnostic masks and classifying the masks. Many works have already identified the bottleneck in open-vocabulary segmentation lies in the mask classification part. We further conducted experiments using ground-truth masks as input, which means the output of the first step is perfect, and the results obtained demonstrate the effectiveness of mask classification. You can refer to [this](https://github.com/hustvl/MaskAdapter/blob/main/mask_adapter/mask_adapter.py). The first stage of training uses ground-truth masks as input, and the evaluation is also conducted with gt masks as input. This is the experimental code for the motivation analysis.

**Comment by ShunZuo-AI** [comment]
Thank you for your reply. I have a question about the MaskAdapter module you proposed. According to the structure diagram of the paper, the input of MaskAdapter is clip feature and gt masks, which seem to be two modalities After designing the module, why do you think it can improve classification performance solely based on scientific intuition? Can you share your ideas for designing this module?

**Comment by ShunZuo-AI** [comment]
What I mean is, I want to reproduce your motivation results again, including mask cropping and mask pooling. How should I modify your code to obtain the results in your image

![Image](https://github.com/user-attachments/assets/dbc6b9ff-0991-4946-ba4e-bd5988939eaf)

**Comment by owl-10** [comment]
> What I mean is, I want to reproduce your motivation results again, including mask cropping and mask pooling. How should I modify your code to obtain the results in your image
> 
> ![Image](https://github.com/user-attachments/assets/dbc6b9ff-0991-4946-ba4e-bd5988939eaf)

For mask pooling,you can replace [mask-adapter](https://github.com/hustvl/MaskAdapter/blob/01fdcb42debdd085f187beca1941992760af7a93/mask_adapter/mask_adapter.py#L404) with  [FC-CLIP](https://github.com/bytedance/fc-clip/blob/2b0bbe213070d44da9182530fa2e826fef03f974/fcclip/fcclip.py#L358).
For mask cropping,you can refer to [zsseg](https://github.com/MendelXu/zsseg.baseline).

**Comment by owl-10** [comment]
> Thank you for your reply. I have a question about the MaskAdapter module you proposed. According to the structure diagram of the paper, the input of MaskAdapter is clip feature and gt masks, which seem to be two modalities After designing the module, why do you think it can improve classification performance solely based on scientific intuition? Can you share your ideas for designing this module?

Compared to Mask Pooling, which uses binary masks (0 and 1) to aggregate mask region features, our Mask-Adapter extracts semantic activation maps from the masks. Mask-Adapter offers several key advantages: (1) Rather than neglecting the background, it aggregates mask embeddings across the entire image, incorporating contextual information to enrich the feature representation; (2) Unlike mask pooling, which only conveys the positional information of target regions, semantic activation maps selectively highlight informative regions relevant to recognition, while suppressing less informative areas, thus enhancing feature discriminability. The structure of Mask-Adapter is simple: we start by patchifying the masks and then pass them through several ConvNeXt blocks to finally output multiple semantic activation maps. We introduce CLIP features branch to fuse more image semantic information.

**Comment by ShunZuo-AI** [comment]
Thank you for your answer to my previous question. I have gained a lot from it. I have another detailed question, which is that you mentioned in your paper that "using ADE20K ground truth masks as input, we evaluate the upper bound of different extraction methods". Have you tested all the images of ADE20K, including the training set and validation set? Or only the test set, does this ADE20k refer to ADEChallengeData2016 or ADE20K2021_17_01?
![Image](https://github.com/user-attachments/assets/9bb689c5-358a-4471-8c93-fc143000f017)

**Comment by owl-10** [comment]
Only use ADEChallengeData2016 validation set for evaluation.

**Comment by ShunZuo-AI** [comment]
Thank you for your reply. Wishing you even better achievements in the future
